### PR TITLE
Extract common abstractions

### PR DESCRIPTION
**Pull request by avdb13**

### CODE REVIEW

File: atrium-common/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+[package]
+name = "atrium-common"
+version = "0.1.0"
+authors = ["sugyan <sugi1982+github@gmail.com>"]
```
Comment (sugyan) [inline atrium-common/Cargo.toml:4]: Your name should be written here.

**Review by sugyan — CHANGES_REQUESTED** [review]
Thanks for your great work!
I think this is a very nice implementation.
I have submitted a change request to your repository, including removing unnecessary dependencies. Please check that much.
https://github.com/avdb13/atrium/pull/1

**Comment by sugyan** [comment]
@avdb13  What about avdb13#1? I'd like to keep dependencies to a minimum and get rid of unnecessary code.

**Comment by avdb13** [comment]
> @avdb13  What about avdb13#1? I'd like to keep dependencies to a minimum and get rid of unnecessary code.

Apologies, I did not know you opened a PR for my fork to suggest changes.

**Comment by avdb13** [comment]
There we go.

**Review by sugyan — APPROVED** [review]
LGTM!

**Pull request merged**